src/diffusers/models/vae.py

```diff
@@ -280,6 +280,253 @@ def custom_forward(*inputs):
         return sample
 
 
+class UpSample(nn.Module):
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+    ) -> None:
+        super().__init__()
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.deconv = nn.ConvTranspose2d(in_channels, out_channels, kernel_size=4, stride=2, padding=1)
+
+    def forward(self, x: torch.FloatTensor) -> torch.FloatTensor:
+        x = torch.relu(x)
+        x = self.deconv(x)
+        return x
+
+
+class MaskConditionEncoder(nn.Module):
```

Comment:
One very very minor point: 

`MaskConditionEncoder` sounds like the matching encoder for `MaskConditionDecoder` but it's not :) 
it is just an encoder to encode image and mask conditions -

Reply:
it's @patrickvonplaten suggestion :D

Reply:
ok then :)